File: app/src/main/java/com/drivingschoolrwandaapp/models/response/RegisterResponse.java
package com.drivingschoolrwandaapp.models.response;

import com.google.gson.annotations.SerializedName;

public class RegisterResponse {

    @SerializedName("success")
    private boolean success;

    @SerializedName("message")
    private String message;

    @SerializedName("error")
    private String error;

    @SerializedName("userId")
    private int userId;

    public boolean isSuccess() {
        return success;
    }

    public String getMessage() {
        return message != null ? message : error;
    }

    public int getUserId() {
        return userId;
    }
}

File: app/src/main/java/com/drivingschoolrwandaapp/models/response/RegisterResponse.kt
package com.drivingschoolrwandaapp.models.response

import com.google.gson.annotations.SerializedName

data class RegisterResponse(
    @get:JvmName("isSuccess") @SerializedName("success") var success: Boolean = false,
    @SerializedName("message") private var _message: String? = null,
    @SerializedName("error") var error: String? = null,
    @SerializedName("userId") var userId: Int = 0
) {
    val message: String?
        get() = _message ?: error
}
